### PR TITLE
feat: implement initial DiT Engine with Master/Scheduler/Engine/Worker/Executor. 

### PR DESCRIPTION
**Pull request by yiming-l21**
xLLM originally focused on supporting LLMs (with core structures and interfaces tailored to LLM workflows), and this PR builds the DiT Engine to enable xLLM’s support for Diffusion-based models like Flux (text-to-image).

- Reconstructed LLM-compatible components that were incompatible with Diffusion logic, including defining core data structures and rewriting processing flows of key classes.
- Aligned the DiT Engine with xLLM’s existing architecture: API Service & Master & Scheduler & Engine & Worker & Executor.

### CODE REVIEW

File: xllm/core/framework/request/dit_request.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ public:
+  DiTRequest(const std::string& request_id,
+             const std::string& x_request_id,
+             const std::string& x_request_time,
+             const DiTRequestState& state,
+             const std::string& service_request_id = "");
+
+  bool finished() const;
+
+  DiTRequestOutput generate_output(DiTForwardOutput dit_output);
```
Comment (yq33victor) [inline xllm/core/framework/request/dit_request.h]: `DiTForwardOutput dit_output` -> `const DiTForwardOutput& dit_output`

File: xllm/core/framework/request/dit_request_state.h [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+};
+
+struct DiTInputParams {
+  std::string prompt;
+
+  std::optional<std::string> prompt_2;
+
+  std::optional<std::string> negative_prompt;
+
+  std::optional<std::string> negative_prompt_2;
```
Comment (yq33victor) [inline xllm/core/framework/request/dit_request_state.h:64]: nit: maybe it's better to add some comments :)

File: xllm/core/framework/dit_model_loader.h [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+#include "core/framework/model/model_args.h"
+#include "core/framework/quant_args.h"
+#include "core/framework/state_dict/state_dict.h"
+#include "core/framework/tokenizer/tokenizer.h"
+#include "core/framework/tokenizer/tokenizer_args.h"
+#include "model_loader.h"
+namespace xllm {
+class DiTFolderLoader {
+ public:
```
Comment (yq33victor) [inline xllm/core/framework/dit_model_loader.h:30]: could we implement `DiTModelLoader` base `class ModelLoader`? like `class HFModelLoader : public ModelLoader` it seems that most of the functions are the same as those in `Class ModelLoader`.

File: xllm/core/framework/request/dit_request_state.h [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+ public:
+  DiTRequestState(DiTInputParams& input_params,
+                  DiTGenerationParams& generation_params,
+                  const DiTOutputFunc& output_func,
+                  const DiTOutputsFunc& outputs_func,
+                  std::optional<Call*> call = std::nullopt)
+      : input_params_(input_params),
+        generation_params_(generation_params),
+        output_func_(output_func),
+        outputs_func_(outputs_func),
```
Comment (yq33victor) [inline xllm/core/framework/request/dit_request_state.h]: nit: 
If possible, we can consider using `std::move(...)`

File: xllm/core/runtime/dit_engine.cpp [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+#include <glog/logging.h>
+#include <sys/sysinfo.h>
+
+#include "core/framework/dit_model_loader.h"
+#include "worker.h"
+
+namespace xllm {
+DiTEngine::DiTEngine(const runtime::Options& options) : options_(options) {
+  const auto& devices = options_.devices();
```
Comment (yq33victor) [inline xllm/core/runtime/dit_engine.cpp:27]: could we create a abstract `class XXXEngine`, include some common functions like: `init_model`, `step`, `get_active_activation_memory ` ... 

The implementations of `DiTEngine` and `VLMEngine` are very similar.

File: xllm/core/runtime/dit_worker.cpp [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+#if defined(USE_NPU)
+  torch::npu::synchronize();
+#elif defined(USE_MLU)
+// TODO(mlu): implement mlu synchronize stream
+#endif
+  COUNTER_ADD(execution_latency_seconds_model, timer.elapsed_seconds());
+
+  DiTForwardOutput output;
+  output.image = hidden_states;
+  return output;
```
Comment (yq33victor) [inline xllm/core/runtime/dit_worker.cpp:112]: nit: std::move(hidden_states)

File: xllm/core/runtime/dit_worker.cpp [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+folly::SemiFuture<folly::Unit> DiTWorker::process_group_test_async() {
+  folly::Promise<folly::Unit> promise;
+  auto future = promise.getSemiFuture();
+  threadpool_.schedule([this, promise = std::move(promise)]() mutable {
+    this->process_group_test_async();
```
Comment (yq33victor) [inline xllm/core/runtime/dit_worker.cpp:119]: recursive call `process_group_test_async ` ?

**Review by yq33victor — APPROVED** [review]
LGTM
